src/storage/disk_manager.c

```diff
@@ -4126,62 +4144,62 @@ DISK_ISVALID
 disk_is_page_sector_reserved_with_debug_crash (THREAD_ENTRY * thread_p, VOLID volid, PAGEID pageid, bool debug_crash)
 {
   PAGE_PTR page_volheader = NULL;
-  DISK_VOLUME_HEADER *volheader;
-  DISK_ISVALID isvalid = DISK_VALID;
-  SECTID sectid;
-  bool old_check_interrupt;
-  int old_wait_msecs;
+  DISK_VOLUME_HEADER *volheader = NULL;
 
-  old_check_interrupt = logtb_set_check_interrupt (thread_p, false);
-  old_wait_msecs = xlogtb_reset_wait_msecs (thread_p, LK_INFINITE_WAIT);
+  bool old_check_interrupt = logtb_set_check_interrupt (thread_p, false);
+  int old_wait_msecs = xlogtb_reset_wait_msecs (thread_p, LK_INFINITE_WAIT);
 
-  if (fileio_get_volume_descriptor (volid) == NULL_VOLDES || pageid < 0)
+  auto exit_routine =
+    [&thread_p, &debug_crash, &page_volheader, &old_check_interrupt, &old_wait_msecs] (DISK_ISVALID isvalid) {
+    assert (isvalid != DISK_INVALID || !debug_crash);
```

Comment:
A bit convoluted with this assert and the subsequent condition (even though function `disk_is_sector_reserved`'s return value does conform to it). It would be clearer if let to the original form + lambda function call.